### PR TITLE
Ignore 'file already exists' errors when recursively making directories

### PR DESCRIPTION
**Pull request by simihartstein**
Addresses issue: Race condition in iModelHost.startup #7452 

Ignore when a folder was already created between the existence check and the attempt to create it.
This would typically only happen if the folder was created out-of-process or in a multi-threaded environment.

### CODE REVIEW

**Review by nick4598 — APPROVED** [review]

File: core/backend/src/IModelJsFs.ts [hunk 1]
```diff
@@ -143,7 +143,14 @@ export class IModelJsFs {
     const parentPath = path.dirname(dirPath);
     if (parentPath !== dirPath)
       IModelJsFs.recursiveMkDirSync(parentPath);
-    IModelJsFs.mkdirSync(dirPath);
+    try {
+      IModelJsFs.mkdirSync(dirPath);
+    } catch (err: any) {
+      // Ignore the error if the folder has been created since the existence check
+      if (err?.code !== 'EEXIST') {
+      throw err;
```
Comment (aruniverse) [inline core/backend/src/IModelJsFs.ts:151]:
```suggestion
        throw err;
```

**Review by aruniverse — APPROVED** [review]
missing rush change

File: core/backend/src/IModelJsFs.ts (hunk 1 above)
Comment (simihartstein) [reply core/backend/src/IModelJsFs.ts:151]: Yes. I didn't have the right node version to run it. Working on it...

Comment (simihartstein) [reply core/backend/src/IModelJsFs.ts]: All resolved. Just an FYI that the linter didn't catch the missing indent or that I used single-quotes. Not sure if that's a problem...

**Comment by aruniverse** [comment]
@mergifyio backport release/4.11.x

**Comment by nick4598** [comment]
@tcobbs-bentley 

Multiple PRs seem to be failing with this [error](https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=3443693&view=logs&j=607f17b7-2b33-5225-6b64-00ff0e2e25e4&t=f5e7265c-6069-5510-b2a5-6d9ff9b2a81c):

>Ld /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/imodeljs-test-app.app/imodeljs-test-app.debug.dylib normal (in target 'imodeljs-test-app' from project 'imodeljs-test-app')
    cd /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app
    /Applications/Xcode-16.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -reproducible -target x86_64-apple-ios17.0-simulator -dynamiclib -isysroot /Applications/Xcode-16.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.1.sdk -O0 -L/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -L/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator -F/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -F/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/PackageFrameworks -F/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator -filelist /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/imodeljs-test-app.build/Debug-iphonesimulator/imodeljs-test-app.build/Objects-normal/x86_64/imodeljs-test-app.LinkFileList -install_name @rpath/imodeljs-test-app.debug.dylib -Xlinker -rpath -Xlinker /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/PackageFrameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/imodeljs-test-app.build/Debug-iphonesimulator/imodeljs-test-app.build/Objects-normal/x86_64/imodeljs-test-app_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode-16.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/imodeljs-test-app.build/Debug-iphonesimulator/imodeljs-test-app.build/Objects-normal/x86_64/imodeljs_test_app.swiftmodule -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Xlinker -alias -Xlinker _main -Xlinker ___debug_main_executable_dylib_entry_point -framework IModelJsNative -framework WebKit -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Intermediates.noindex/imodeljs-test-app.build/Debug-iphonesimulator/imodeljs-test-app.build/Objects-normal/x86_64/imodeljs-test-app_dependency_info.dat -o /Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/imodeljs-test-app.app/imodeljs-test-app.debug.dylib
ld: warning: ignoring file '/Volumes/Builds/vsts_work_a/25/s/test-apps/display-test-app/ios/imodeljs-test-app/build/DerivedData/Build/Products/Debug-iphonesimulator/IModelJsNative.framework/IModelJsNative': found architecture 'arm64', required architecture 'x86_64'
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_IModelJsHost", referenced from:
       in ViewController.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Could you take a look? Probably related to the removal of x64 on mac.

**Review by DanRod1999 — APPROVED** [review]

**Review by tcobbs-bentley — APPROVED** [review]

**Comment by tcobbs-bentley** [comment]
> @tcobbs-bentley
> 
> Multiple PRs seem to be failing with this [error](https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=3443693&view=logs&j=607f17b7-2b33-5225-6b64-00ff0e2e25e4&t=f5e7265c-6069-5510-b2a5-6d9ff9b2a81c):
...
> Could you take a look? Probably related to the removal of x64 on mac.

https://github.com/iTwin/itwinjs-core/pull/7456

**Comment by tcobbs-bentley** [comment]
The iOS problem is now fixed in master, and I updated this PR from master to pull the fix.

**Pull request merged**